packages/safe-core-sdk/src/Safe.ts

```diff
@@ -353,75 +383,92 @@ class Safe {
+   * @param params - The transaction params
+   * @param options - The transaction optional properties
    * @returns The Safe transaction ready to be signed
    * @throws "Invalid owner address provided"
    * @throws "Address provided is already an owner"
    * @throws "Threshold needs to be greater than 0"
    * @throws "Threshold cannot exceed owner count"
    */
-  async getAddOwnerTx(ownerAddress: string, threshold?: number): Promise<SafeTransaction> {
+  async getAddOwnerTx(
+    { ownerAddress, threshold }: AddOwnerTxParams,
+    options?: SafeTransactionOptionalProps
+  ): Promise<SafeTransaction> {
     const safeTransaction = await this.createTransaction({
       to: this.getAddress(),
       value: '0',
-      data: await this.#ownerManager.encodeAddOwnerWithThresholdData(ownerAddress, threshold)
+      data: await this.#ownerManager.encodeAddOwnerWithThresholdData(ownerAddress, threshold),
+      ...options
     })
     return safeTransaction
   }
 
   /**
    * Returns the Safe transaction to remove an owner and optionally change the threshold.
    *
-   * @param ownerAddress - The address of the owner that will be removed
-   * @param threshold - The new threshold
+   * @param params - The transaction params
+   * @param options - The transaction optional properties
    * @returns The Safe transaction ready to be signed
    * @throws "Invalid owner address provided"
    * @throws "Address provided is not an owner"
    * @throws "Threshold needs to be greater than 0"
    * @throws "Threshold cannot exceed owner count"
    */
-  async getRemoveOwnerTx(ownerAddress: string, threshold?: number): Promise<SafeTransaction> {
+  async getRemoveOwnerTx(
+    { ownerAddress, threshold }: RemoveOwnerTxParams,
+    options?: SafeTransactionOptionalProps
+  ): Promise<SafeTransaction> {
     const safeTransaction = await this.createTransaction({
       to: this.getAddress(),
       value: '0',
-      data: await this.#ownerManager.encodeRemoveOwnerData(ownerAddress, threshold)
+      data: await this.#ownerManager.encodeRemoveOwnerData(ownerAddress, threshold),
+      ...options
     })
     return safeTransaction
   }
 
   /**
    * Returns the Safe transaction to replace an owner of the Safe with a new one.
    *
-   * @param oldOwnerAddress - The old owner address
-   * @param newOwnerAddress - The new owner address
+   * @param params - The transaction params
+   * @param options - The transaction optional properties
    * @returns The Safe transaction ready to be signed
    * @throws "Invalid new owner address provided"
    * @throws "Invalid old owner address provided"
    * @throws "New address provided is already an owner"
    * @throws "Old address provided is not an owner"
    */
-  async getSwapOwnerTx(oldOwnerAddress: string, newOwnerAddress: string): Promise<SafeTransaction> {
+  async getSwapOwnerTx(
+    { oldOwnerAddress, newOwnerAddress }: SwapOwnerTxParams,
+    options?: SafeTransactionOptionalProps
```

Comment:
Do you think it makes sense to have a separate interface/type for options excluding the operation? For example, for the owner management methods, you don't want to change the operation
```
export type CallTransactinoOptions = Omit<SafeTransactionOptionalProps, 'operation'>
```

Reply:
Makes sense. I will update that